### PR TITLE
[MU4] Fix GH#10653: Crash converting to whole note

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Resolves: #10653

Same issue with (and originally reported for) MU3

### CODE REVIEW

**Comment by Tantacrul** [comment]
Cool! Just tested it and it seems to be solving the crash. @RomanPudashkin / @Eism - if you are happy with this, it would be nice to merge (removes a P0).

Thanks @Jojo-Schmitz !

**Comment by Jojo-Schmitz** [comment]
What about merging this?

**Comment by Jojo-Schmitz** [comment]
Any particular reason __not__ to merge this?

**Comment by Tantacrul** [comment]
@bkunda - when you get a moment, can you test this and assuming all's good, recommend to merge?

I can't quite manage it at the mo.

**Comment by bkunda** [comment]
> @bkunda - when you get a moment, can you test this and assuming all's good, recommend to merge?
> 
> I can't quite manage it at the mo.

The logs for this run have apparently expired so I can no longer download any artefacts.

**Comment by Jojo-Schmitz** [comment]
Downside of letting PRs sit for such a long time...
Rebased and force-pushed now.

File: src/engraving/libmscore/edit.cpp [hunk 1]
```diff
@@ -4830,6 +4830,9 @@ static EngravingItem* findLinkedVoiceElement(EngravingItem* e, Staff* nstaff)
     Measure* measure = segment->measure();
     Measure* m       = score->tick2measure(measure->tick());
     Segment* s       = m->findSegment(segment->segmentType(), segment->tick());
+    if (!s) {
```
Comment (RomanPudashkin) [inline src/engraving/libmscore/edit.cpp:4833]: This looks very suspicious. Where do these null pointers come from? It looks like this PR fixes the consequence of the problem, not the cause

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/edit.cpp:4833]: Well, I don't quite understand the question, obviously this one comes from `findSegment()`.

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/edit.cpp:4833]: That in turn used `findSegmentR()`
```
//---------------------------------------------------------
//   findSegmentR
//    Search for a segment of type st at measure relative
//    position t.
//---------------------------------------------------------

Segment* Measure::findSegmentR(SegmentType st, const Fraction& t) const
{
    Segment* s;
    if (t > (ticks() * Fraction(1, 2))) {
        // search backwards
        for (s = last(); s && s->rtick() > t; s = s->prev()) {
        }
        while (s && s->prev() && s->prev()->rtick() == t) {
            s = s->prev();
        }
    } else {
        // search forwards
        for (s = first(); s && s->rtick() < t; s = s->next()) {
        }
    }
    for (; s && s->rtick() == t; s = s->next()) {
        if (s->segmentType() & st) {
            return s;
        }
    }
    return 0;
}
```

Comment (RomanPudashkin) [reply src/engraving/libmscore/edit.cpp:4833]: Why does it return null in this case? It usually indicates some other, more complicated problem

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/edit.cpp:4833]: Dunno. The method does on occasions and on purpose return a `nullptr`. So callers should take care of those and not relying on it to never do that.

Comment (Jojo-Schmitz) [reply src/engraving/libmscore/edit.cpp:4833]: If it were illegal, an assert would be due in that `findSegmentR()`

**Comment by RomanPudashkin** [comment]
This PR fixes one symptom of the problem (crash in one scenario), but it doesn't fix the problem completely (removed element in the selection). For example, if I cherry-pick the changes from the PR, I can't select anything in the note input bar after the steps from the issue (due to the corrupted selection list). I guess we will be able to reproduce this crash in other scenarios as well (even with these changes)

**Comment by Jojo-Schmitz** [comment]
Ah, I see.
Don't see the fix for that though :-(
Still won't invalidate my pointer checking in this PR, would it?

**Comment by RomanPudashkin** [comment]
Let's try this fix https://github.com/musescore/MuseScore/pull/13183

It prevents removed elements from being added to the selected element list

**Comment by Jojo-Schmitz** [comment]
That's fine, and of course better than my attempt here, but #13183 still allows Measure::findSegmentR(`)` to return 0, which still may cause a crash whenever the return falue of this is dereferenced. Should be prevented by an `assert()` at least.

**Pull request closed without merging**